This patch fixes several issues with WebGL on NVIDIA/X11:

    * Work around issue with WebKit-supplied external context
      See https://bugs.webkit.org/show_bug.cgi?id=261653

    * Fix segmentation fault with libepoxy;

    * Set m_anglePlatform for X11.

--- webkitgtk-2.42.0/Source/WebCore/platform/graphics/angle/PlatformDisplayANGLE.cpp~	2023-08-09 09:49:45.616316800 +0000
+++ webkitgtk-2.42.0/Source/WebCore/platform/graphics/angle/PlatformDisplayANGLE.cpp	2023-10-01 19:33:34.270286923 +0000
@@ -74,9 +74,6 @@
         return m_angleSharingGLContext;
 
     ASSERT(m_angleEGLDisplay != EGL_NO_DISPLAY);
-    auto sharingContext = sharingGLContext();
-    if (!sharingContext)
-        return EGL_NO_CONTEXT;
 
     EGLint configAttributes[] = {
         EGL_RENDERABLE_TYPE, EGL_OPENGL_ES2_BIT,
@@ -95,7 +92,6 @@
     if (numberConfigsReturned != 1)
         return EGL_NO_CONTEXT;
 
-    GLContext::ScopedGLContextCurrent scopedCurrent(*sharingContext);
     EGLint contextAttributes[] = {
         EGL_CONTEXT_CLIENT_VERSION, 2,
         EGL_EXTERNAL_CONTEXT_ANGLE, EGL_TRUE,
--- webkitgtk-2.42.0/Source/WebCore/platform/graphics/texmap/GraphicsContextGLTextureMapperANGLENicosia.cpp~	2023-09-26 10:51:52.915360944 +0000
+++ webkitgtk-2.42.0/Source/WebCore/platform/graphics/texmap/GraphicsContextGLTextureMapperANGLENicosia.cpp	2023-09-26 10:52:08.999307697 +0000
@@ -27,7 +27,7 @@
 #include "GraphicsContextGLTextureMapperANGLE.h"
 
 #if ENABLE(WEBGL) && USE(TEXTURE_MAPPER) && USE(NICOSIA)
-#include <epoxy/gl.h>
+#include <GL/gl.h>
 
 namespace WebCore {
 
--- webkitgtk-2.42.0/Source/WebCore/platform/graphics/x11/PlatformDisplayX11.cpp-	2023-09-26 12:55:29.301613100 +0000
+++ webkitgtk-2.42.0/Source/WebCore/platform/graphics/x11/PlatformDisplayX11.cpp	2023-09-26 12:53:59.971122612 +0000
@@ -69,6 +69,9 @@
 PlatformDisplayX11::PlatformDisplayX11(Display* display)
     : m_display(display)
 {
+#if ENABLE(WEBGL)
+    m_anglePlatform = EGL_PLATFORM_X11_KHR;
+#endif
 }
 
 #if PLATFORM(GTK)
@@ -76,6 +79,9 @@
     : PlatformDisplay(display)
     , m_display(display ? GDK_DISPLAY_XDISPLAY(display) : nullptr)
 {
+#if ENABLE(WEBGL)
+    m_anglePlatform = EGL_PLATFORM_X11_KHR;
+#endif
 }
 #endif
 
